src/StateMachine/StateMachine.ts

```diff
@@ -127,13 +133,24 @@ export class StateMachine {
         );
       }
     } catch (error) {
+      /*
+       * Returning to the global handler here only makes sense if we didn't already made that,
+       * meaning that it could only be done in the first recussion. There's tests covering this scenarion
+       * in tests/States.spec.ts
+       */
+      if (fromState !== "entry" && recursions < 1) {
```

Comment:
In which part of the code is this variable updated ?

Reply:
https://github.com/mediarain/voxa/pull/166/files#diff-50197b557be628917c89e4d26537925aR108